specutils/analysis/flux.py

```diff
@@ -169,20 +169,29 @@ def _compute_equivalent_width(spectrum, continuum=1, regions=None,
     if hasattr(calc_spectrum, 'mask') and calc_spectrum.mask is not None:
         mask = calc_spectrum.mask
         calc_spectrum = Spectrum1D(flux=calc_spectrum.flux[~mask],
-                              spectral_axis=calc_spectrum.spectral_axis[~mask])
+                                   spectral_axis=calc_spectrum.spectral_axis[~mask])
+    else:
+        mask = None
 
     if continuum == 1:
         continuum = 1*calc_spectrum.flux.unit
 
     spectral_axis = calc_spectrum.spectral_axis
 
-    dx = (np.abs(spectral_axis[-1] - spectral_axis[0]))
+    if regions is not None:
+        cont_spec = Spectrum1D(flux=continuum*np.ones(spectral_axis.shape),
+                               spectral_axis=calc_spectrum.spectral_axis)
+        cont_flux = _compute_line_flux(cont_spec, mask_interpolation=mask_interpolation)
+    else:
+        cont_flux = continuum * (np.abs(spectral_axis.bin_edges[-1] - spectral_axis.bin_edges[0]))
 
-    line_flux = _compute_line_flux(spectrum, regions,
-                                   mask_interpolation=mask_interpolation)
+    if mask is not None:
+        line_flux = _compute_line_flux(spectrum, regions, mask_interpolation=mask_interpolation)
+    else:
+        line_flux = _compute_line_flux(calc_spectrum, mask_interpolation=mask_interpolation)
```

Comment:
This part confuses me at the moment. `calc_spectrum` should have already dealt with regions and masks at this point, so I'm not sure why we're reverting back to using `spectrum` when it seems the line flux ought to be calculated from `calc_spectrum`. It looks as though previously, the mask wasn't being considered?

Reply:
Frankly I did not understand how the mask was taken care of in the current version either, and have not tracked that down. In my first version I also used `calc_spectrum` throughout; this broke the test for `equivalent_width(spectrum_masked, mask_interpolation=FluxConservingResampler)`
as it returned the same EW as the `LinearInterpolatedResampler` one. However with other example data like the SDSS-1323 spectrum from the main documentation I could not find any differences between the two methods either, and in light of your comments maybe the results should indeed be the same?

Reply:
Hmm. The case of using `FluxConservingResampler` _should_ be different, albeit very close to the case of the `LinearInterpolatedResampler`, so I would not expect them to be the same. But it's not clear why using `calc_spectrum` vs `spectrum` broke the tests.